domain/resource/doc.go

```diff
@@ -1,8 +1,34 @@
 // Copyright 2024 Canonical Ltd.
 // Licensed under the AGPLv3, see LICENCE file for details.
 
-// Package resource provides the domain types for a resource. The application
-// domain is the primary domain for creation and handling of a resource. The
-// types here are to ensure that the resource is handled correctly and that
-// the resource is correctly represented in the domain.
+// Package resource provides the domain types for handling charm resources once
```

Comment:
I think this sentence should be expanded to say explicitly that resources are created along with the application in the application domain, but all other operations are handled through this domain. It does sort of already say this, but I don't think I would get it if I was reading it for the first time.